### DISCUSSION TITLE
Computed does not work in nested store

### DISCUSSION THREAD

**Discussion by p0wl90** [vuejs/pinia#2640] [Help and Questions] [answered]
### Reproduction

https://play.pinia.vuejs.org/#eNqlVsGO2jAQ/RUrlwUJ4rKsttI2INpqD63UbdVtb7mExIB3EzuynYCE8u8d2yRxCLC0vQX7zfPMe3g8e+9jnvtlQbwHL5CxoLlCkqgiR2nE1rPQUzL05iGjWc6FQntUSPItomykP56IVCRBFVoJnqEbH0vFBZG+kjchC1nMmVQoA/SzXkezOngwrDeZYXC2LaUGBNimA4fDD0WyPI0UgV8IBZvp/LekbI3Uhhw4kDkblTQyi/rUw9KAcYW2XLxCwDDAEGs4loVSnKFFnNL4FQpt8vQt4VORLYmArCYgwDNRSHE0CbAN+0uKu9uW4+62S5LQcn468AHt9618Xc6qCrCOdChYEaWISqughX1nBMUbEr+iQS3AJdLZDKo9RV2j++wdcY+4++gO9VIYa4/tdJzbUrXhBah27HJTzTV26q//MrNDcL2VbdiRLg7feRtbUGPieZorjNP3ruHrGdXZvcqmjh+uUe7Ne9smpwH8+727QPKWXedCjUJnec/b5sK6xp1P8rR5Lr53k3qM19y1ADtd1Bt5tqePsyj3XyRn8ALsNTg8bEDjh2P0il7LKaORXgk9bL59IrPxUvCtJALiQ29UQxfwnOCElIrzVI6jnNqwjVK5fMA4ThjgE5LSUoASCrM8w72Yxb1/709wSpcYDsKUJWTXPQYixgnJrmGvoYt3/uTOf29Yy+mBNNOsmrQKWQWyKAlv04quj0SJeZbTFLTNFYW3qyNOlKZ8+9WsKVGQJkXj/on1F7mzWf+A55KIkjhlqUisibLbj89PZAffzWbGkyIF9IXNn0TytNA5WtingiWQtoMz2X4xHsPl/CUfd4owWRelEzVqGLyR+fOF0tt0p/7UUbEZBUDBZnhIyIoyYp/7emowfyaYFxqQIKsR0moX7nABaZihguwMzI4P7Qwyc7kHN/Zq3IzQYIhmc5utO3E0/QEOC5j9MR9Mhh9aYP9GzZqsBpbW5fLLKC0Aoy+zpREwRgkGBbmw0QneCuCVjgkZxm7/VJvI1Ch7Hbevg+7gxypoqlMadB60Swp0n4Ve/S1Pv3pXAftf6bZXZ9brtMPDv/REx3MjetuHsDajhqdTg1k96O1VfwBWxM3V

### Steps to reproduce the bug

1. Click on "Set mainStore.nestedNumber to 42"

### Expected behavior

mainStore.isNestedNumberOne is false

### Actual behavior

mainStore.isNestedNumberOne is true

### Additional information

- I tried various things like wrapping isNestedNumberOne in computed() in the main store but I can't get it to work and it seems like a bug to me
- There are two variations in the Pinia Playground that work as expected

**Comment by posva** [comment, marked as answer]
The issue is about reactivity with Vue. By doing

```ts
return {
    nestedNumber: useNested().nestedNumber,
    isNestedNumberOne: useNested().isNestedNumberOne,
    mainNumber,
    isMainNumberOne,
  };
```

You are reading the numbers from `nestedNumber` and the boolean from `isNestedNumberOne`. They are just values, not reactive wrappers (Ref), so they aren't reactive anymore: https://vuejs.org/guide/essentials/reactivity-fundamentals.html#limitations-of-reactive

Here is a [fixed version of what you are trying to do](https://play.pinia.vuejs.org/#eNqlVtFu2jAU/RUrL4CUxmNUndQFxDb1YZPWTe32lpeQGHCb2FHsBCSUf9+1TRKHBMq2FxTsc4/vPce+9sH5lGVeWRDn3vFFlNNMIkFkkaEkZJt54EgROIuA0TTjuUQHVAjyPaTMVR+PREgSowqtc56ikYeF5DkRnhSjgAUs4kxIlAL6WY2jeR08ntSTTDNY04ZSAXxs0oHF4Y8kaZaEksA/hPztbPFbULZBckuOHEivjUoa6kG16nFozLhEO56/QsDExxCrOVaFlJyhZZTQ6BUKbfL0DOFjka5IDllNQYBnIpHkaOpjE/aXFLfvW47b912SmJaL4cB7dDi08nU5q8rHKtKiYEWYICqMggb2gxEUbUn0isa1AJdI53Oodoi6RvfZO+KecPfRHepVrq09tdNybkfllheg2qnLTTXX2Km+/svMDsH1VrZhJ7pYfOdtbEGNiedprjBOnbuGr2dUZ/Yqmzp+2EbZJ+9tm6wG8O/n7gLJW3adC9UKneU9b5sN6xp3Pslh82x87yT1GK85az62uqjjOqan36Rh5r0IzuAGOChwcJyAxg/LqBE1llFGQzUSOFh/e0SkN6uc7wTJIT5w3Bq6hOsEx6SUnCfiJsyoCdtKmYl7jKOYAT4mCS1zUEJilqW4F7O88+68KU7oCsNCmLKY7LvLQMRNTNJr2Gvo8p03vfU+aNZydiRNFasirQJWgSxSwN20ppsTUSKeZjQBbTNJ4e7qiBMmCd9902MyL0iTonZ/YPxF7E3WP+G6JHlJrLJkmG+INNMPz49kD9/NZMrjIgH0hcknInhSqBwN7HPBYkjbwulsv2qP4XD+Eg97SZioi1KJajU0Xsv85ULpbbozb2ap2DwFQMHm8RCTNWXEXPf1q0FvJngvNKCcrF2k1C5gP7twcJ/IugZDMvppQfYabB4R7Utkbq8wHpkDMnLReILmC5Oz/e5ougQs6TPzZzGeTj62wP65mje5jQ2tzeWVYVIARh1pQ5PDYypnUJYNcwd4K4BXKiZgGNtdVG5DXaPo9d2+DqqPn6qgqIY06FxrlxToXg69+luefvVvvvJgC1kimU0F5cecjST8QslUoFAgCv+KDPo9dC8jhJDwiaDcFZdbo4cw8d0+rvfP2ErBRceNYRCjyfFgDDTZQafPNNyappWj4e0IqEePZjvVH4ue+s0=) but it duplicates the state in two stores, avoid doing that, use a computed to not duplicate the source of truth.
